pallets/federated-authority-observation/src/lib.rs

```diff
@@ -179,38 +176,49 @@ pub mod pallet {
 		#[allow(clippy::useless_conversion)]
 		pub fn reset_members(
 			origin: OriginFor<T>,
-			council_authorities: Vec<(T::AccountId, MainchainMember)>,
-			technical_committee_authorities: Vec<(T::AccountId, MainchainMember)>,
+			council_authorities: BoundedVec<(T::AccountId, MainchainMember), T::CouncilMaxMembers>,
+			technical_committee_authorities: BoundedVec<
+				(T::AccountId, MainchainMember),
+				T::TechnicalCommitteeMaxMembers,
+			>,
 		) -> DispatchResultWithPostInfo {
 			ensure_none(origin)?;
 
-			let (council_account_ids, council_mainchain_members): (Vec<_>, Vec<_>) =
-				council_authorities.into_iter().unzip();
-			let (technical_committee_account_ids, technical_committee_mainchain_members): (
+			let (mut council_members, council_mainchain_members): (Vec<_>, Vec<_>) =
+				council_authorities.clone().into_iter().unzip();
+			let (mut technical_committee_members, technical_committee_mainchain_members): (
 				Vec<_>,
 				Vec<_>,
-			) = technical_committee_authorities.into_iter().unzip();
+			) = technical_committee_authorities.clone().into_iter().unzip();
 
-			// Prepare Council members
-			let mut council_members: BoundedVec<T::AccountId, T::CouncilMaxMembers> =
-				BoundedVec::try_from(council_account_ids.clone())
-					.map_err(|_| Error::<T>::TooManyMembers)?;
+			let council_members_len = council_members.len() as u32;
+			let technical_committee_members_len = technical_committee_members.len() as u32;
 
+			// ---- Council ----
 			// Make sure an empty set of members is not allowed
 			ensure!(!council_members.is_empty(), Error::<T>::EmptyMembers);
 			council_members.sort();
 
+			// Make sure there are not duplicated members
+			ensure!(
```

Comment:
Haven't seen this macro before - neat!

Reply:
Checking this - is it intentional to stop block production if applying new members fails due to input validation? This applies to all the `ensure!` statements in this call

I think better behaviour would be to emit an error event rather than returning an error